src/cascadia/QueryExtension/AzureLLMProvider.cpp

```diff
@@ -0,0 +1,228 @@
+            jsonContent.SetNamedValue(L"temperature", WDJ::JsonValue::CreateNumberValue(0.7));
+            jsonContent.SetNamedValue(L"frequency_penalty", WDJ::JsonValue::CreateNumberValue(0));
+            jsonContent.SetNamedValue(L"presence_penalty", WDJ::JsonValue::CreateNumberValue(0));
+            jsonContent.SetNamedValue(L"top_p", WDJ::JsonValue::CreateNumberValue(0.95));
+            jsonContent.SetNamedValue(L"stop", WDJ::JsonValue::CreateStringValue(L"None"));
+            const auto stringContent = jsonContent.ToString();
+            WWH::HttpStringContent requestContent{
+                stringContent,
+                WSS::UnicodeEncoding::Utf8,
+                L"application/json"
+            };
+
+            request.Content(requestContent);
+
+            // Send the request
+            try
+            {
+                const auto response = _httpClient.SendRequestAsync(request).get();
+                // Parse out the suggestion from the response
+                const auto string{ response.Content().ReadAsStringAsync().get() };
+                const auto jsonResult{ WDJ::JsonObject::Parse(string) };
+                if (jsonResult.HasKey(errorString))
+                {
+                    const auto errorObject = jsonResult.GetNamedObject(errorString);
+                    message = errorObject.GetNamedString(messageString);
+                }
+                else
+                {
+                    if (_verifyModelIsValidHelper(jsonResult))
+                    {
+                        const auto choices = jsonResult.GetNamedArray(L"choices");
+                        const auto firstChoice = choices.GetAt(0).GetObject();
+                        const auto messageObject = firstChoice.GetNamedObject(messageString);
+                        message = messageObject.GetNamedString(contentString);
+                        isError = false;
+                    }
+                    else
+                    {
+                        message = RS_(L"InvalidModelMessage");
+                    }
+                }
+            }
+            catch (...)
+            {
+                message = RS_(L"UnknownErrorMessage");
+            }
+        }
+
+        // Also make a new entry in our jsonMessages list, so the AI knows the full conversation so far
+        WDJ::JsonObject responseMessageObject;
+        responseMessageObject.Insert(roleString, WDJ::JsonValue::CreateStringValue(L"assistant"));
+        responseMessageObject.Insert(contentString, WDJ::JsonValue::CreateStringValue(message));
+        _jsonMessages.Append(responseMessageObject);
+
+        co_return winrt::make<AzureResponse>(message, isError);
+    }
+
+    bool AzureLLMProvider::_verifyModelIsValidHelper(const WDJ::JsonObject jsonResponse)
+    {
+        const auto model = jsonResponse.GetNamedString(L"model");
+        bool modelIsAccepted{ false };
+        for (const auto acceptedModel : acceptedModels)
```

Comment:
i'm sure if we didn't just use a `string[]` we could do some sort of smarter `.contains()`, but also, _eh_. We accept like 5 models, this is fine.